converter/internal/promtailconvert/internal/build/service_discovery.go

```diff
@@ -0,0 +1,95 @@
+package build
+
+import (
+	"github.com/grafana/agent/converter/internal/common"
+	"github.com/grafana/agent/converter/internal/prometheusconvert"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	prom_discover "github.com/prometheus/prometheus/discovery"
+)
+
+func (s *ScrapeConfigBuilder) AppendSDs() {
+	sdConfigs := toDiscoveryConfig(s.cfg)
+	if len(sdConfigs) == 0 {
+		return
+	}
+
+	pb := prometheusconvert.NewPrometheusBlocks()
+	targets := prometheusconvert.AppendServiceDiscoveryConfigs(pb, sdConfigs, common.LabelForParts(s.globalCtx.LabelPrefix, s.cfg.JobName))
+	pb.AppendToFile(s.f)
+
+	for _, target := range targets {
+		s.allTargetsExps = append(s.allTargetsExps, target["__expr__"])
+	}
+
+	s.diags.AddAll(prometheusconvert.ValidateServiceDiscoveryConfigs(sdConfigs))
+}
+
+func toDiscoveryConfig(cfg *scrapeconfig.Config) prom_discover.Configs {
```

Comment:
let's turn the promtail SD configs into a type compatible with the current prometheus API